node/node.go

```diff
@@ -223,33 +196,84 @@ func (n *Node) KeepRegistered() {
 	n.registerCh <- struct{}{}
 }
 
-// Multicast start the mulicast ping
-func (n *Node) Multicast(ctx context.Context, multicastAddress string) error {
+// Multicast start the multicast ping
+func (n *Node) Multicast(ctx context.Context, multicastAddress string) <-chan error {
+	errCh := make(chan error, 1)
+
 	n.mu.Lock()
+	n.mc = &multicast.Multicast{Address: multicastAddress}
 	n.ipv4 = localIPv4()
 	n.ipv6 = localIPv6()
 
+	// Stop existing ticker if it exists
+	if n.ticker != nil {
+		n.ticker.Stop()
+	}
+
 	if n.SendInterval.Seconds() == float64(0) {
 		n.SendInterval = 1 * time.Second // default to 1 second
 	}
-
+	n.ticker = time.NewTicker(n.SendInterval)
 	n.mu.Unlock()
 
-	// Encode node to be sent via multicast
-	n.mu.Lock()
-	n.mc = &multicast.Multicast{Address: multicastAddress}
-	n.mu.Unlock()
-	if err := n.mc.Send(ctx, n.SendInterval, n.Encode()); err != nil {
-		return err
+	// Create send function
+	send := func() error {
+		// Check to see if IPv4 or 6 changed
+		currentIPv4 := localIPv4()
+		currentIPv6 := localIPv6()
+
+		if !n.ipv4.Equal(currentIPv4) || !n.ipv6.Equal(currentIPv6) {
+			// Multicast a deregister of previous node
+			n.mu.Lock()
+			n.Action = DeregisterAction
+			n.mu.Unlock()
```

Comment:
Why unlock here?

Reply:
node's Encode method now locks the same mutex.